File: .make-it/prototype/stubs/io/sentry/sqlite/SentrySQLiteDriver.java
package io.sentry.sqlite;

import androidx.sqlite.SQLiteDriver;

/**
 * Stub mirroring the real SentrySQLiteDriver.create() contract from sentry-java PR #5466: static
 * create(SQLiteDriver): SQLiteDriver, idempotent for an already-SentrySQLiteDriver delegate.
 */
public final class SentrySQLiteDriver implements SQLiteDriver {
  private final SQLiteDriver delegate;

  private SentrySQLiteDriver(SQLiteDriver delegate) {
    this.delegate = delegate;
  }

  @Override
  public Object open(String fileName) {
    return delegate.open(fileName);
  }

  public static SQLiteDriver create(SQLiteDriver delegate) {
    System.out.println("    >> SentrySQLiteDriver.create() wrapping " + delegate.getClass().getName());
    // idempotency: re-wrapping a SentrySQLiteDriver is a no-op
    if (delegate instanceof SentrySQLiteDriver) {
      return delegate;
    }
    return new SentrySQLiteDriver(delegate);
  }
}

File: plugin-build/src/test/kotlin/io/sentry/sqlite/SentrySQLiteDriver.kt
package io.sentry.sqlite

import androidx.sqlite.SQLiteDriver

/**
 * Minimal stub of `io.sentry.sqlite.SentrySQLiteDriver` so ASM can resolve the `INVOKESTATIC
 * io/sentry/sqlite/SentrySQLiteDriver.create` emitted by [SetDriverMethodVisitor]. The `@JvmStatic
 * create(SQLiteDriver)` shape here mirrors the SDK contract the visitor depends on.
 */
class SentrySQLiteDriver private constructor(private val delegate: SQLiteDriver) : SQLiteDriver {
  companion object {
    @JvmStatic fun create(delegate: SQLiteDriver): SQLiteDriver = SentrySQLiteDriver(delegate)
  }
}
